x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/TrainedModelStatsService.java

```diff
@@ -146,6 +153,12 @@ void updateStats() {
         if (clusterState == null || statsQueue.isEmpty() || stopped) {
             return;
         }
+
+        boolean isInUpgradeMode = MlMetadata.getMlMetadata(clusterState).isUpgradeMode();
+        if (isInUpgradeMode) {
+            return;
```

Comment:
debug logging? Might be helpful for troublshooting issues where stats are mysteriously not being persisted.